week3/average/kirikaza/Average.hs

```diff
@@ -0,0 +1,24 @@
+module Main where
+
+import Control.Monad (forM_)
+import Data.Function (on)
+import qualified Data.Sequence as Seq
+import Data.Sequence ((<|), Seq(..))
+
+main :: IO ()
+main = forM_ [4,2] $ \window -> do
+  print $ movingAvgList window [1,5,3,8,7,9,6]
+
+movingAvgList :: Int -> [Int] -> [Float]
+movingAvgList windowSize values = map avg $ formWindows windowSize values
+
+formWindows :: Int -> [value] -> [Seq value]
```

Comment:
`a` is usually used instead of `value`; named type variables (i.e. not `a`, `b`, `f`, `m`, etc) are used pretty rarely, usually with multiparameter type classes or in situations like this one: http://hackage.haskell.org/package/req-0.1.0/docs/Network-HTTP-Req.html#v:req